lib/ui/painting.dart

```diff
@@ -3850,8 +3853,20 @@ class FragmentShader extends Shader {
         'FragmentShader floatUniforms size: ${floatUniforms.length} must match given shader uniform count: $_uniformFloatCount.');
     }
     _update(floatUniforms);
+    _hashCode = hashValues(_spirvHash, hashList(floatUniforms));
```

Comment:
How expensive are these precalculated hash values? Can this be done lazily? (If we want to implement `==` correctly then we'd need to save the buffers any way.